src/Badger.ts

```diff
@@ -0,0 +1,103 @@
+import { warning, info } from '@actions/core'
+import Cache from "./util/Cache";
+import _ from 'lodash'
+import { replaceAll } from "./util/string";
+import Badge from "./util/badge";
+
+export const findReferences = (source: string) => 
+  _.uniq(source.match(/{{.+?}}/g)).map(variable => variable.match(/{{(.+?)}}/)[1])
+
+export default class Badger {
+  constructor(private variables: Cache<string>) {}
+
+  public applyPrefix(body: string, prefix?: string) {
```

Comment:
Wondering if it's worth changing the public interface for `Badger` 🤔 

I'm thinking it might be worth making it more high level and instead exposing a single method;

```ts
public apply(body: string, prefix: string, suffix: string, badges: string[]): string
```

This method would then do everything in one which will also make it easier to support running on PR updates;

```ts
github.onPullRequest(PullRequestAction.OPENED, (context: Context, octokit: Octokit, helper: PullRequestHelper) => {
  const body = helper.getPRDescripition()
  const badger = new Badger(makeCache(context))
  
  info('Generating badges...')
  const updatedBody = badger.apply(body, getInput('prefix'), getInput('suffix'), getBadgeConfigs())
 
  info('Updating PR description...')
  helper.setPRDescription(updatedBody)
})
```